### PR TITLE
feat: user terminal

### PR DESCRIPTION
**Pull request by sashahonchar**

### CODE REVIEW

File: src/terminal/index.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.watcher = chokidar.watch(this.baseDir + 'output', {
+      awaitWriteFinish: {
+        stabilityThreshold: 500,
+        pollInterval: 100,
+      },
+    });
+  }
+
+  private exec(command: string): Promise<TerminalResult> {
+    const echoStatus = 'echo $? > ' + this.baseDir + 'exitStatus';
```
Comment (kgilpin) [inline src/terminal/index.ts:68]: How about using `fs.join` here?

File: src/agent/appMapAgentRuby.ts [hunk 2]
```diff
@@ -68,29 +74,25 @@ export default class AppMapAgentRuby implements AppMapAgent {
...
-          output: true,
-        });
+        const { stderr, exitCode } = await execCommand(path, 'bundle', ['install']);
 
         if (exitCode !== 0) {
           error = stderr;
         }
       }
     } catch (e) {
+      console.log(e);
```
Comment (kgilpin) [inline src/agent/appMapAgentRuby.ts:95]: Can/should this be directed to a VSCode output?

**Comment by kgilpin** [comment]
@sashahonchar you'll probably want to clean up the history before merging.

File: src/agent/appMapAgentRuby.ts (hunk 2 above)
Comment (sashahonchar) [reply src/agent/appMapAgentRuby.ts:95]: Error message will be displayed on Quickstart webview:
https://github.com/applandinc/vscode-appland/blob/7868c2069be47775bbeeb3bdf7dc7c62eeba3b54/src/agent/appMapAgentRuby.ts#L96

File: src/terminal/index.ts (hunk 1 above)
Comment (sashahonchar) [reply src/terminal/index.ts]: Done https://github.com/applandinc/vscode-appland/pull/294/commits/6eb052092ee5a995529d3f1fff62ca54ae0df846

**Comment by sashahonchar** [comment]
> @sashahonchar you'll probably want to clean up the history before merging.

Done. I squashed 'wip' commits into this one https://github.com/applandinc/vscode-appland/pull/294/commits/fd7d8ec02c444498a2b98ab162d5b35b52bf94af

**Review by kgilpin — CHANGES_REQUESTED** [review]
I think it's good that we did this exercise but we need to hold off on replacing the existing flow with this flow for all situations. 

Also, chokidar should be removed and replaced with the built in VSCode file watcher. Native code is not allowed in VSCode extensions.

**Comment by torchello** [comment]
Closing this for now.

**Pull request closed without merging**